### PR TITLE
Add a way to consume flow types for the end users

### LINKED ISSUE #55: There is no easy way to consume flow typings
**Do you want to request a *feature* or report a *bug*?**

This is likely a bug

**What is the current behavior?**

Using `import NavigationPrompt from 'react-router-navigation-prompt';` within flow codebase results in `NavigationPrompt` being typed as `any`

**If the current behavior is a bug, please provide the steps to reproduce.**

See current behavior

**What is the expected behavior?**

It can have already defined typings from `src/index.js`.

For the moment, this can be achieved by using

- `import NavigationPrompt from 'react-router-navigation-prompt/src';`
- copy-pasting content of the `src/index.js` into `es/index.js.flow` with the help of something like `patch-package`
- users can add following line to the `options` in `.flowconfig`: 
    `module.name_mapper='^react-router-navigation-prompt$' -> '<PROJECT_ROOT>/node_modules/react-router-navigation-prompt/src/index.js'`

neither of these looks anyhow good from lib user experience

**Please post the output of `npm list react react-router react-router-dom react-router-navigation-prompt` (with redactions if needed).**

```
├── flow-bin@0.84.0 
├── react@16.8.6 
├─┬ react-router-dom@4.3.1 
│ └── react-router@4.3.1 
└── react-router-navigation-prompt@1.8.8 
```

### PR DESCRIPTION
**Pull request by 102**
fixes #55

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,6 +51,7 @@
     "babel-preset-flow": "^6.23.0",
     "babel-preset-react": "^6.24.1",
     "concurrently": "^3.5.1",
+    "copy-webpack-plugin": "^4.6.0",
```
Comment (102) [inline package.json]: Should both `package-lock.json` & `yarn.lock` be updated?

File: webpack.config.js [hunk 2]
```diff
@@ -7,6 +8,17 @@ module.exports = {
     filename: 'index.js',
     libraryTarget: 'commonjs2' // THIS IS THE MOST IMPORTANT LINE! :mindblow: I wasted more than 2 days until realize this was the line most important in all this guide.
   },
+  plugins: [
+    new CopyPlugin([
+      // copy the main source file to the `es/index.js.flow` to add
+      // a way to consume flow types for the end users
+      {
+        from: 'src/index.js',
+        to: 'index.js.flow',
```
Comment (102) [inline webpack.config.js]: this is neither the best solution (as it produces the extra files in the repo, though it wouldn't affect end users bundle as `.flow` files are only for `flow` type checking), nor the bulletproof solution (e.g. it wouldn't work anyhow if source code will be splitted in the future), but it is the only solution that came into my mind for the moment

File: es/index.js.flow [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+  beforeCancel?: Function,
+  beforeConfirm?: Function,
+  children: (data: {isActive: bool, onCancel: Function, onConfirm: Function}) => React$Element<*>,
+  match: Match,
+  history: RouterHistory,
+  location: Location,
+  renderIfNotActive: bool,
+  when: bool | (Location, ?Location) => bool,
+  disableNative: bool,
+  allowGoBack: bool,
```
Comment (102) [inline es/index.js.flow]: These typings should also be updated to be more useful (as it is done in ts typings):

```
  allowGoBack?: boolean;
  renderIfNotActive?: boolean;
  disableNative?: boolean;
```

but I think it would be better to do it in the separate PR

File: es/index.js.flow [hunk 4]
```diff
@@ -0,0 +1,201 @@
+/* @flow */
+import React from 'react';
+import {withRouter} from 'react-router-dom';
+import type {HistoryAction, Location, Match, RouterHistory} from 'react-router-dom';
+
+declare type PropsT = {
+  afterCancel?: Function,
```
Comment (102) [inline es/index.js.flow]: `Function` & `Object` types are deprecated in the recent versions of flow

File: package.json (hunk 1 above)
Comment (ZacharyRSmith) [reply package.json]: Ideally, yes.  Would you mind PR'ing that?  I'll go ahead and merge this.

**Comment by ZacharyRSmith** [comment]
now live as v1.8.9
